daemon/openrazer_daemon/hardware/keyboards.py

```diff
@@ -397,6 +397,32 @@ def _close(self):
         self._battery_manager.close()
 
 
+class RazerBlackWidowV3ProWireless(RazerBlackWidowV3ProWired):
+    """
+    Class for the Razer BlackWidow v3 Pro 2.4 Ghz Wireless
+    """
+    USB_PID = 0x025C
+    METHODS = ['get_device_type_keyboard', 'set_wave_effect', 'set_static_effect', 'set_spectrum_effect',
+               'set_reactive_effect', 'set_none_effect', 'set_breath_random_effect', 'set_breath_single_effect', 'set_breath_dual_effect',
+               'set_custom_effect', 'set_key_row', 'get_game_mode', 'set_game_mode', 'get_macro_mode', 'set_macro_mode',
+               'get_macro_effect', 'set_macro_effect', 'get_macros', 'delete_macro', 'add_macro',
+               'set_starlight_random_effect', 'set_starlight_single_effect', 'set_starlight_dual_effect',
+               'set_charge_effect', 'set_charge_colour', 'get_battery', 'is_charging', 'set_low_battery_threshold']
```

Comment:
Why does it now have the `'set_ripple_effect', 'set_ripple_effect_random_colour',` functions the wired version has? Other than that it seems to match the base class `RazerBlackWidowV3ProWired` so if those two also work there's no need to override `METHODS` since it's set in the base class

Reply:
There is no ripple effect in the wireless version. I think I don't get your point 😅

Reply:
Are you sure ripple doesn't work? It's a software effect built on the matrix / custom frame functionality so there shouldn't be a reason it doesn't work apart from wrong `EVENT_FILE_REGEX` setup

Reply:
Ah now I get it.
I'll add the ripple effect and check if it works. Could take some time though since I have a lot to do this week.

Reply:
So this is the only thing that remains to be addressed? (and rebasing to fix the conflicts of course)

Reply:
![image](https://github.com/openrazer/openrazer/assets/1645737/0c7f8b14-41a9-4479-b379-8d5064854ff2)

With just rebasing.

Reply:
@z3ntu So I've added back `'set_ripple_effect', 'set_ripple_effect_random_colour'` to the wireless class and nothing happens from Polychromatic UI when I select « Ondulation » (ripple). What should I check about `EVENT_FILE_REGEX` ?

Reply:
Looks like the regex might be wrong? `/dev/input/by-id/usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-event-kbd` is what i have locally

Reply:
Updated the regex with `EVENT_FILE_REGEX = re.compile(r'.*Razer_Razer_BlackWidow_V3_Pro(_\d+)?(-if01)?-event-kbd')` and verified it matches:
```
search_dir /dev/input/by-id/
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if02-event-mouse
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if02-event-mouse DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if02-mouse
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if02-mouse DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if01-event-mouse
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if01-event-mouse DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if01-mouse
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-if01-mouse DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-event-kbd
search_dir /dev/input/by-id/ event_file usb-Razer_Razer_BlackWidow_V3_Pro_000000000002-event-kbd MATCH
search_dir /dev/input/by-id/ event_file usb-Logitech_USB_Receiver-if02-event-mouse
search_dir /dev/input/by-id/ event_file usb-Logitech_USB_Receiver-if02-event-mouse DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-if01-event-mouse
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-if01-event-mouse DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-event-kbd
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-event-kbd DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-event-if01
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-event-if01 DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Cosair_Corsair_VOID_RGB_USB_Gaming_Headset_00000000-event-if03
search_dir /dev/input/by-id/ event_file usb-Cosair_Corsair_VOID_RGB_USB_Gaming_Headset_00000000-event-if03 DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Lenovo_ThinkPad_Thunderbolt_3_Dock_USB_Audio_000000000000-event-if03
search_dir /dev/input/by-id/ event_file usb-Lenovo_ThinkPad_Thunderbolt_3_Dock_USB_Audio_000000000000-event-if03 DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Logitech_USB_Receiver-if02-mouse
search_dir /dev/input/by-id/ event_file usb-Logitech_USB_Receiver-if02-mouse DONT MATCH
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-if01-mouse
search_dir /dev/input/by-id/ event_file usb-Generic_Virtual_HID_00000002-if01-mouse DONT MATCH
```

Yet ripple effect still does nothing. I've also tried over wired connection to the keyboard and it was not working either.

Reply:
Opened new PR at https://github.com/openrazer/openrazer/pull/2231